client/src/GamemasterInGame/PlayerListComponents/PlayerNoteInput.tsx

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import { usePlayerNotes } from "../../store/useStore";
+import { Button, Dialog, Flex, TextArea } from "@radix-ui/themes";
+
+interface PlayerNoteInputProps {
+  player: string;
+  children: React.ReactNode;
+  note?: string;
+}
+
+export function PlayerNoteInput({
+  player,
+  children,
+  note = "",
+}: PlayerNoteInputProps) {
+  const [, playerNotesLoading, , setPlayerNote] = usePlayerNotes();
+  const [newNote, setNewNote] = React.useState(note);
+
+  return (
+    <Dialog.Root onOpenChange={() => setNewNote(note)}>
+      <Dialog.Trigger disabled={playerNotesLoading}>{children}</Dialog.Trigger>
+
+      <Dialog.Content className="m-2">
+        <Dialog.Title className="capitalize">
+          <Flex justify="between">
+            <label className="flex-1" htmlFor="note-input">
+              {player}: Notes
+            </label>
+            <Dialog.Close>
+              <Button
+                type="reset"
+                variant="surface"
+                onClick={() => setPlayerNote(player, "")}
```

Comment:
I actually don't think this should close.  Because it's so destructive you should have to save it too